arch/cortex-m/samv71-hal/src/xdmac/channel.rs

```diff
@@ -0,0 +1,510 @@
+    }
+
+    /// Resumes destination requests for the channel.
+    ///
+    /// # Safety
+    /// This is a read-modify-write operation that uses global XDMAC registers. Be very careful
+    /// with that if you share the Channels between threads/IRQs.
+    pub fn resume_write(&mut self) {
+        self.xdmac_registers_ref()
+            .gws
+            // Safety: This is safe, because channel's ID must be valid for a Channel to exist.
+            // Also, channel bits are correctly masked with old value, as this is an R-M-W operation.
+            // Channel bitmask must be negated, as this operation is supposed to clear the channel bit.
+            .modify(|r, w| unsafe { w.bits(r.bits() & !self.channel_bitmask()) });
+    }
+
+    /// Resumes read and write operations at the same time.
+    pub fn resume_read_and_write(&mut self) {
+        self.xdmac_registers_ref()
+            .grwr
+            // Safety: This is safe, because channel's ID must be valid for a Channel to exist.
+            .write(|w| unsafe { w.bits(self.channel_bitmask()) });
+    }
+
+    /// Returns channel's ID.
+    pub fn id(&self) -> usize {
+        self.id
+    }
+
+    /// Returns a reference to channel's registers.
+    #[inline(always)]
+    fn channel_registers_ref(&self) -> &ChannelRegisters {
+        // Safety: This is safe, as the address of the register is guaranteed to be valid by Xdmac.
+        unsafe { &*self.channel_registers }
+    }
+
+    /// Returns a reference to XDMAC's registers.
+    #[inline(always)]
+    fn xdmac_registers_ref(&self) -> &RegisterBlock {
+        // Safety: This is safe, as the address of XDMAC register is guaranteed to be valid by Xdmac.
+        unsafe { &*Self::XDMAC_REGISTERS }
+    }
+
+    /// Returns channel's bitmask (`1` shifted by `n` bits, where `n` is channel's ID)
+    /// This function will return valid value as long, as channel's ID is also valid.
+    #[inline(always)]
+    fn channel_bitmask(&self) -> u32 {
+        1 << self.id
+    }
+
+    /// Returns `true` if channel's bit is set in specified value.
+    /// The value should usually be a register's content.
+    #[inline(always)]
+    fn is_channels_bit_set(&self, value: u32) -> bool {
+        value & self.channel_bitmask() != 0
+    }
+
+    /// Transforms Channel into a type with different state.
+    ///
+    /// This is a helper function that reduces state transition boilerplate.
+    ///
+    /// # Parameters
+    /// * `channel` - Channel instance to be consumed and transformed.
+    ///
+    /// # Returns
+    /// Transformed Channel instance.
+    const fn transform<NewState: State>(channel: Channel<NewState>) -> Self {
+        Self {
+            channel_registers: channel.channel_registers,
+            id: channel.id,
+            status_reader: channel.status_reader,
+            _state: PhantomData,
+        }
+    }
+
+    /// Pointer to XDMAC's registers.
+    const XDMAC_REGISTERS: *const RegisterBlock = XDMAC::PTR;
```

Comment:
Move to the top maybe?

Reply:
it's private, shouldn't it be below public members?

Reply:
resonable